src/execute.rs

```diff
@@ -94,6 +96,55 @@ pub fn execute_mint(
         .add_attribute("color_map", format!("{:?}", new_color_map)))
 }
 
+pub fn execute_change_url(
+    deps: DepsMut,
+    info: MessageInfo,
+    env: Env,
+    position: u16,
+    url: String,
+) -> Result<Response, ContractError> {
+    let owner_check = get_owner(deps.as_ref(), env, position);
+    let owner;
+
+    match owner_check {
+        None => {
+            return Err(ContractError::DoesNotExist {});
+        }
+        Some(_) => owner = owner_check.unwrap(),
+    }
+
+    if owner != info.sender {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    let token_id = position.to_string();
+    let token = tokens().load(deps.storage, &token_id)?;
+
+    let extension = token.extension;
+
+    let updated_extension = PixelExtension {
+        pixel_colors: extension.pixel_colors,
+        url: url.clone(),
+    };
+
+    let updated_token = TokenInfo::<PixelExtension> {
+        owner: Addr::unchecked(owner),
+        approvals: token.approvals,
+        token_uri: token.token_uri,
+        extension: updated_extension,
+    };
+
+    tokens().update(deps.storage, &token_id, |existing| match existing {
```

Comment:
I think this can be more elegantly done with 
```
tokens().replace(deps.storage, &token_id, Some(&new_token), Some(&token))?;
```
Source: [XYZ with a similar usecase](https://github.com/collectxyz/collectxyz-nft-contract/blob/5e9e37445b217993880b7553391c5b62977b2179/src/execute.rs#L224)

Reply:
ooooh yeah this is way nicer